### PR TITLE
Add override_host_experimental config and related fields in LoadBalancingConfig

### PR DESCRIPTION
**Pull request by sanjaypujare**

### CODE REVIEW

**Review by markdroth — CHANGES_REQUESTED** [review]

File: grpc/service_config/service_config.proto [hunk 1]
```diff
@@ -415,6 +415,9 @@ message XdsClusterResolverLoadBalancingPolicyConfig {
 
   // xDS LB policy. Will be used as the child config of the xds_cluster_impl LB policy. 
   repeated LoadBalancingConfig xds_lb_policy = 2;
+
+  // override_host_status in CDS includes values we support for stateful session affinity.
+  bool override_host_enabled = 3;
```
[the remark this answers is not in the document]
Comment (sanjaypujare) [reply grpc/service_config/service_config.proto:420]: Okay, got it.

File: grpc/service_config/service_config.proto [hunk 2]
```diff
@@ -527,6 +530,13 @@ message LeastRequestLocalityLoadBalancingPolicyConfig {
   uint64 choice_count = 1;
 }
 
+
+// Configuration for the override_host LB policy.
+message OverrideHostLoadBalancingPolicyConfig {
```
[the remark this answers is not in the document]
Comment (sanjaypujare) [reply grpc/service_config/service_config.proto:532]: done

File: grpc/service_config/service_config.proto [hunk 3]
```diff
@@ -527,6 +527,15 @@ message LeastRequestLocalityLoadBalancingPolicyConfig {
...
+  // override_host_status in CDS includes values we support for stateful session affinity.
```
Comment (markdroth) [inline grpc/service_config/service_config.proto:533]: This comment should just say something like "if false, the LB policy will be a no-op."  It does not need to reference anything xDS-specific.

Comment (sanjaypujare) [reply grpc/service_config/service_config.proto]: done

**Comment by sanjaypujare** [comment]
I believe I have addressed all the comments. Is this waiting for the gRFC A55 to be merged first?

**Comment by markdroth** [comment]
Yes, this needs to wait for the gRFC to be ready.

Note that there are pending comments in the gRFC that will require changes in this PR before it can be merged.

**Comment by sanjaypujare** [comment]
> Note that there are pending comments in the gRFC that will require changes in this PR before it can be merged.

Made changes to reflect the latest design of the proto

File: grpc/service_config/service_config.proto [hunk 4]
```diff
@@ -622,6 +642,8 @@ message LoadBalancingConfig {
       [json_name = "xds_wrr_locality_experimental"];
     LeastRequestLocalityLoadBalancingPolicyConfig least_request_experimental = 17
       [json_name = "least_request_experimental"];
+    OverrideHostLoadBalancingPolicyConfig override_host_experimental = 18
```
Comment (markdroth) [inline grpc/service_config/service_config.proto:645]: Please move this up to line 632.  This policy is not deprecated, so it doesn't belong in this section (see comment on line 633).

(There are some existing entries here that also need to move up, which I'm fixing in #115.)

**Review by markdroth — APPROVED** [review]
Looks great!

**Comment by sanjaypujare** [comment]
Is it okay to merge now or should I wait (for gRFC to merge etc)?

**Comment by markdroth** [comment]
I think we can go ahead and merge this.  If we wind up needing subsequent changes later, we can make a separate PR.

**Pull request merged**